src/ape/managers/project.py

```diff
@@ -451,35 +451,44 @@ def contracts(self) -> Dict[str, ContractType]:
 
     def __getattr__(self, attr_name: str) -> ContractContainer:
         """
-        Get a contract container from an existing contract type or dependency
-        name using ``.`` access.
+        Get a contract container from an existing contract type in
+        the local project using ``.`` access.
+
+        **NOTE**: To get a dependency contract, use
+        :py:attr:`~ape.managers.project.ProjectManager.dependencies`.
 
         Usage example::
 
             from ape import project
 
             contract = project.MyContract
 
+        Raises:
+            AttributeError: When the given name is not a contract in the project.
+
         Args:
-            attr_name (str): The name of the contract or dependency.
+            attr_name (str): The name of the contract in the project.
 
         Returns:
             :class:`~ape.contracts.ContractContainer`
         """
 
-        contracts = self.load_contracts()
-        if attr_name in contracts:
-            contract_type = contracts[attr_name]
-        elif attr_name in self.dependencies:
-            contract_type = self.dependencies[attr_name]  # type: ignore
-        else:
-            # Fixes anomaly when accessing non-ContractType attributes.
-            # Returns normal attribute if exists. Raises 'AttributeError' otherwise.
-            return self.__getattribute__(attr_name)  # type: ignore
+        if attr_name in self.contracts:
+            source_id = self.contracts[attr_name].source_id
+            if source_id:
+                if source_id.startswith(".cache"):
+                    raise AttributeError(
+                        "Cannot access dependency contracts from root project manager. "
+                        f"Please use `project.dependencies['dependency_name'].{attr_name}`."
+                    )
 
-        return ContractContainer(  # type: ignore
-            contract_type=contract_type,
-        )
+                return ContractContainer(  # type: ignore
+                    contract_type=self.contracts[attr_name],
+                )
```

Comment:
should this be dedented to the right one? `source_id` is not a requirement to load the type